src/defargs.l

```diff
@@ -501,9 +517,54 @@ CPPC  "/\/"
 <CopyArgRound,CopyArgRound2,CopyArgSharp,CopyArgCurly>.  {
 					  *yyextra->copyArgValue += *yytext;
 					}
-<ReadTypeConstraint>[,)]               {
-                                          unput(*yytext);
-                                          BEGIN(yyextra->lastExtendsContext);
+<ReadAnnotation>"("                    {
+                                          ++yyextra->argRoundCount;
+                                       }
+<ReadAnnotation>")"                    {
+                                          if (--yyextra->argRoundCount == 0)
+                                          {
+                                            BEGIN(yyextra->readAnnotationContext);
+                                          }
+                                       }
+<ReadAnnotation>{B}+                   {
+                                          if (yyextra->argRoundCount == 0)
+                                          {
+                                            BEGIN(yyextra->readAnnotationContext);
+                                          }
+                                       }
+<ReadAnnotation>"\""                   {
+                                          BEGIN(ReadStringLiteral);
+                                       }
+<ReadAnnotation>.                      {
+                                          // skip it for now
+                                       }
+<ReadStringLiteral>"\\"                {
+                                          BEGIN(ReadEscape);
+                                       }
```

Comment:
Why not combine this with the rule `<ReadEscape>.` to `<ReadStringLiteral>"\\."` (like it is also done for `<CopyArgString>` )?

Reply:
Okay. Good idea.

Reply:
I found the "\\." slightly misleading for escape sequences because they can be longer than one character, e.g. for unicode ``\uXXXX``. So I went with ``<ReadStringLiteral>"\\\""`` which I think is more on spot for the purpose of that rule. Anyway, just a matter of taste I guess.

Reply:
Probably yes, but I don't really see the problem with the `\uXXXX` as here the `\\.` rule would eat the `\u` and the normal `.` rule `X`s. I think both solutions would be valid in this case.

Reply:
Yes, sure both would work. That's why I wrote it's a matter of taste. I just found the `\\\"` rule more explicit for it's purpose.